ruby-challenges.rb

```diff
@@ -10,16 +10,72 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def locater array, string
+    array.select do |x|
+        if x.include?(string)
+            x
+        end
+   end
+end
+
+p locater(beverages_array, letter_o)
+p locater(beverages_array, letter_t)
+
+
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
+
+def sort_states hash
+   hash.values.flatten.sort
+end
+
+p sort_states(us_states)
+
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
+
+class Bike
+attr_accessor :model, :wheels, :current_speed, :speed
+
+    def initialize(model)
+        @model = model
+        @wheels = 2
+        @current_speed = 0
+        @speed = speed
```

Comment:
recommend removing the speed instance variable on Line 50 since it was never used.